src/bitbox.rs

```diff
@@ -0,0 +1,189 @@
+    Keypath, PairedBitBox,
+};
+use bitcoin::{
+    bip32::{DerivationPath, ExtendedPubKey, Fingerprint},
+    psbt::Psbt,
+};
+use regex::Regex;
+use std::str::FromStr;
+
+pub struct BitBox {
+    pub network: bitcoin::Network,
+    pub display_xpub: bool,
+    pub client: PairedBitBox<TokioRuntime, bitbox_api::usb::Transport>,
+}
+
+impl std::fmt::Debug for BitBox {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BitBox").finish()
+    }
+}
+
+impl BitBox {
+    pub async fn connect() -> Result<Self, Error> {
+        let noise_config = Box::new(bitbox_api::NoiseConfigNoCache {});
+        let bitbox = bitbox_api::BitBox::<
+            bitbox_api::runtime::TokioRuntime,
+            bitbox_api::usb::Transport,
+        >::from(bitbox_api::usb::get_any_bitbox02().unwrap(), noise_config)
+        .await?;
+        let pairing_bitbox = bitbox.unlock_and_pair().await?;
+        Ok(BitBox {
+            display_xpub: false,
+            network: bitcoin::Network::Bitcoin,
+            client: pairing_bitbox.wait_confirm().await?,
+        })
+    }
+
+    pub fn with_network(mut self, network: bitcoin::Network) -> Self {
+        self.network = network;
+        self
+    }
+
+    pub fn display_xpub(mut self, value: bool) -> Self {
+        self.display_xpub = value;
+        self
+    }
+}
+
+#[async_trait]
+impl HWI for BitBox {
+    fn device_kind(&self) -> DeviceKind {
+        DeviceKind::BitBox
+    }
+
+    async fn get_version(&self) -> Result<super::Version, HWIError> {
+        Err(HWIError::UnimplementedMethod)
+    }
+
+    async fn is_connected(&self) -> Result<(), HWIError> {
+        Err(HWIError::UnimplementedMethod)
+    }
+
+    async fn get_master_fingerprint(&self) -> Result<Fingerprint, HWIError> {
+        let fg = self
+            .client
+            .root_fingerprint()
+            .await
+            .map_err(|e| HWIError::Device(e.to_string()))?;
+        Ok(Fingerprint::from_str(&fg).map_err(|e| HWIError::Device(e.to_string()))?)
+    }
+
+    async fn get_extended_pubkey(&self, path: &DerivationPath) -> Result<ExtendedPubKey, HWIError> {
+        let fg = self
+            .client
+            .btc_xpub(
+                if self.network == bitcoin::Network::Bitcoin {
+                    pb::BtcCoin::Btc
+                } else {
+                    pb::BtcCoin::Tbtc
+                },
+                &Keypath::from(path),
+                pb::btc_pub_request::XPubType::Xpub,
+                self.display_xpub,
+            )
+            .await
+            .map_err(|e| HWIError::Device(e.to_string()))?;
+        Ok(ExtendedPubKey::from_str(&fg).map_err(|e| HWIError::Device(e.to_string()))?)
+    }
+
+    async fn register_wallet(
+        &self,
+        name: &str,
+        policy: &str,
+    ) -> Result<Option<[u8; 32]>, HWIError> {
+        let policy = extract_script_config_policy(&policy)?;
+        self.client
+            .btc_register_script_config(
```

Comment:
you should check if it is registered already (using `btc_is_script_config_registered()` and only do this if it isn't.